bin/aws-configure

```diff
@@ -39,72 +38,55 @@ To add a new profile, you MUST specify following environment variables:
 
 You can optionally specify '<PROFILE>_AWS_DEFAULT_REGION' to determine the default region for this profile.
 
-If no PROFILE is specified, ${0##*/} will defer to the value specified in AWS_PROFILE.
-
 SUBCOMMANDS:
     is-configured PROFILE     Returns '0' if the profile is configured. Otherwise, it returns '1'.
 DOC
 }
 
 function setup_profile() {
-  profile="$1"
-  profile_env="$(echo "${profile^^}" | tr '-' '_')"
-
-  enforce_requirements
+  profile="${1}"
```

Comment:
There's no need to add the `{ }` here; they're only required when the positional parameter number is longer than one digit (or when needed to distinguish between variable names with the same prefix). However, if always using the braces is a style choice, I think that's great.

And while we're at it, the quotes aren't necessary either. In general double quotes are needed to  suppress word-splitting and globbing when doing parameter expansion, but that doesn't occur during assignment. If the first argument to `setup_profile` were a literal asterisk:
```bash
setup_profile '*'
```
then `profile` would have the value of a literal asterisk regardless of whether you do
```bash
profile="${1}"
```
or
```bash
profile=$1
```
However, when you _reference_ `$profile`, you had better quote it, or it will expand to whatever matches the glob:
```bash
echo $profile   # prints the non-hidden directory entries
echo "$profile" # prints a literal asterisk
```

Maybe this seems like a small point, but I think it's valuable to know that quoting in the assignment here won't protect later references to the value.